manageiq-operator/pkg/controller/manageiq/manageiq_controller.go

```diff
@@ -0,0 +1,317 @@
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+var log = logf.Log.WithName("controller_manageiq")
+var currentAppName string
+/**
+* USER ACTION REQUIRED: This is a scaffold file intended for the user to modify with their own Controller
+* business logic.  Delete these comments after modifying this file.*
+ */
+
+// Add creates a new Manageiq Controller and adds it to the Manager. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func Add(mgr manager.Manager) error {
+	return add(mgr, newReconciler(mgr))
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager) reconcile.Reconciler {
+	return &ReconcileManageiq{client: mgr.GetClient(), scheme: mgr.GetScheme()}
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New("manageiq-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to primary resource Manageiq
+	err = c.Watch(&source.Kind{Type: &miqv1alpha1.Manageiq{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return err
+	}
+
+	// TODO(user): Modify this to be the types you create that are owned by the primary resource
+	// Watch for changes to secondary resource Pods and requeue the owner Manageiq
+	err = c.Watch(&source.Kind{Type: &appsv1.Deployment{}}, &handler.EnqueueRequestForOwner{
+		IsController: true,
+		OwnerType:    &miqv1alpha1.Manageiq{},
+	})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// blank assignment to verify that ReconcileManageiq implements reconcile.Reconciler
+var _ reconcile.Reconciler = &ReconcileManageiq{}
+
+// ReconcileManageiq reconciles a Manageiq object
+type ReconcileManageiq struct {
+	// This client, initialized using mgr.Client() above, is a split client
+	// that reads objects from the cache and writes to the apiserver
+	client client.Client
+	scheme *runtime.Scheme
+}
+
+// Reconcile reads that state of the cluster for a Manageiq object and makes changes based on the state read
+// and what is in the Manageiq.Spec
+// TODO(user): Modify this Reconcile function to implement your Controller logic.  This example creates
+// a Pod as an example
+// Note:
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcileManageiq) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.Info("Reconciling Manageiq")
+
+	// Fetch the Manageiq instance
+	miqInstance := &miqv1alpha1.Manageiq{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, miqInstance)
+
+	if errors.IsNotFound(err) {
+		err = CleanUpOrchestratedDeployments(miqInstance, r)
```

Comment:
Is `err` unused in this context? Can we remove this assignment or should we be handling this error?